shared/constants/teams.tsx

```diff
@@ -67,39 +67,26 @@ export const initialMemberInfo = Object.freeze<Types.MemberInfo>({
 })
 
 export const rpcDetailsToMemberInfos = (
-  allRoleMembers: RPCTypes.TeamMembersDetails
+  members: Array<RPCTypes.TeamMemberDetails>
 ): Map<string, Types.MemberInfo> => {
   const infos: Array<[string, Types.MemberInfo]> = []
-  const types: Types.TeamRoleType[] = ['reader', 'writer', 'admin', 'owner', 'bot', 'restrictedbot']
-  const typeToKey: Types.TypeMap = {
-    admin: 'admins',
-    bot: 'bots',
-    owner: 'owners',
-    reader: 'readers',
-    restrictedbot: 'restrictedBots',
-    writer: 'writers',
-  }
-  types.forEach(type => {
-    const key = typeToKey[type]
-    // @ts-ignore strict
-    const members: Array<RPCTypes.TeamMemberDetails> = (allRoleMembers[key] || []) as any
-    members.forEach(({fullName, joinTime, status, username}) => {
-      infos.push([
+  members.forEach(({fullName, joinTime, status, username, role}) => {
+    const maybeRole = teamRoleByEnum[role]
+    infos.push([
+      username,
+      {
+        fullName,
+        joinTime: joinTime || undefined,
+        status: rpcMemberStatusToStatus[status],
+        type: !maybeRole || maybeRole === 'none' ? 'reader' : maybeRole,
```

Comment:
can `maybeRole === 'none'` happen? are we risking showing wrong if we change it to reader?